kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -470,14 +473,37 @@ protected void handleTopicMetadataRequest(KafkaHeaderAndRequest metadataHar,
         //      e.g. <topic1, {persistent://public/default/topic1-partition-0,...}>
         //   1. no topics provided, get all topics from namespace;
         //   2. topics provided, get provided topics.
-        CompletableFuture<Map<String, List<TopicName>>> pulsarTopicsFuture = new CompletableFuture<>();
+        CompletableFuture<Map<String, List<TopicName>>> pulsarTopicsFuture;
 
         if (topics == null || topics.isEmpty()) {
             // clean all cache when get all metadata for librdkafka(<1.0.0).
             KafkaTopicManager.clearTopicManagerCache();
-            // get all topics
-            getAllTopicsAsync(pulsarTopicsFuture);
+            // get all topics, filter by permissions.
+            pulsarTopicsFuture = getAllTopicsAsync().thenApply((allTopicMap) -> {
+                final Map<String, List<TopicName>> topicMap = new ConcurrentHashMap<>();
+                allTopicMap.forEach((topic, list) -> {
+                   list.forEach((topicName ->
+                           authorize(AclOperation.DESCRIBE, Resource.of(ResourceType.TOPIC, topicName.toString()))
+                           .whenComplete((authorized, ex) -> {
```

Comment:
It's same as my previous comment. There's no need to authorize `DESCRIBE` operation here.

Reply:
Sorry it's my fault, `DESCRIBE` should be authorized. Solve this comment.